argo_cd/datadog_checks/argo_cd/check.py

```diff
@@ -0,0 +1,29 @@
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheck
+
+from .metrics import APPLICATION_METRICS
+
+
+class ArgoCdCheck(OpenMetricsBaseCheck):
```

Comment:
```suggestion
class ArgoCdCheck(OpenMetricsBaseCheckV2):
```
if you want to use the new Openmetrics implementation (better performance, accurate metric behavior to Prometheus)